vignettes/parallel.Rmd

```diff
@@ -0,0 +1,218 @@
+
+* If you are launching many tasks, and know the maximum memory your task needs, then you can specify `memory_per_process` to tell the cluster about that. The cluster will then  avoid allocating too many of your tasks to the same node, if the combined memory needed by those tasks will exceed what the node has. This can't really be guaranteed, unless everyone agrees to set `memory_per_process`, but it should help in the common case where your own tasks might be stacked up on a node together.
+
+## Running on Particular nodes
+
+At present, the nodes on the new cluster are all very similar to each other. This may change over time, and there may be some nodes, or queues of nodes, that are more suitable for your tasks than others.
+
+## Choosing a queue
+
+At present, the new cluster only has one queue for general use, called `AllNodes` containing, as it says, all the available compute nodes. At other times though, during workshops for example we have run a `Training` queue, with strict limits, to ensure we've had capacity to demonstrate cluster use in a live setting.
+
+It also may be necessary in the future to partition the set of nodes, either by their capabilities if that becomes significant to some users, or by which research group might have purchased them, or to allow a particular group more protected access for a period.
+
+Use `hipercow_cluster_info` to see what queues are available, and then pass one of those into the `queue` parameter for `hipercow_resources`, to ensure tasks run on a particular queue.
+
+## Choosing nodes
+
+Even more rarely, you may have a particular node you want to run on. In the past, for instance, we have had specific nodes with specific hardware (such as very large RAM or large disks). Also occasionally, Wes or Rich might want to rerun tasks on a specific node if we want to replicate a failure that occurred on it, and diagnose whether there is a problem on that node.
+
+The list of nodes is provided by `hipercow_cluster_config()`, and you can specify a vector of strings for the node names that you are happy for your tasks to run on, as the `requested_nodes` parameter for `hipercow_resources`.
+
+# Task time limits and scheduling
+
+Our clusters over the years have essentially be run on the basis of good will, rather than having too many limits over how long tasks can run for, or how much resource they can use. For our fairly small department, this is a nice way to work, meaning you can usually get the resources you need, even if they are quite demanding for a period of time. Usage fluctuates depending on deadlines and the development cycle of projects. It is relatively rare that many people or projects have demanding needs at the same time, such that capacity of the cluster becomes problematic. 
+
+That said, Hipercow offers a few options for limiting how long your tasks can run for, specifying when your tasks can run on the cluster, and also allows you to politely allow other tasks to take priority over yours. 
```

Comment:
```suggestion
That said, `hipercow` offers a few options for limiting how long your tasks can run for, specifying when your tasks can run on the cluster, and also allows you to politely allow other tasks to take priority over yours. 
```

Reply:
The benefit of doing this is that we hope fast jobs will be queued ahead of jobs that don't specify a time.  We'll need to fiddle with this of course